### PR TITLE
Expose an attribute "self" in the reaction summary with the list of o…

### PR DESCRIPTION
**Pull request by nickvergessen**
…wn reactions

cc @Ivansss

### CODE REVIEW

File: lib/Controller/ChatController.php [hunk 1]
```diff
@@ -516,6 +501,45 @@ public function receiveMessages(int $lookIntoFuture,
...
+			// Add the self part also to potential parent elements
+			if (isset($parentMap[$messageId])) {
+				foreach ($parentMap[$messageId] as $mid) {
+					if (isset($messages[$commentIdToIndex[$mid]])) {
+						$messages[$commentIdToIndex[$mid]]['parent']['reactions']['self'] = $reactions;
+					}
+				}
+			}
+		}
+
```
Comment (vitormattos) [inline lib/Controller/ChatController.php:542]: Will be good put this in a method and reduce the responsibility of controller. Maybe, a good place to put the new method is in ReactionManager.

args: $message, $messages
return: $messages

File: lib/Chat/ReactionManager.php [hunk 2]
```diff
@@ -169,6 +169,20 @@ public function retrieveReactionMessages(Room $chat, Participant $participant, i
 		return $reactions;
 	}
 
+	/**
+	 * @param Participant $participant
+	 * @param array $messageIds
+	 * @return array[]
+	 * @psalm-return array<int, string[]>
+	 */
+	public function getReactionsForMessages(Participant $participant, array $messageIds): array {
```
Comment (vitormattos) [inline lib/Chat/ReactionManager.php:178]:
```suggestion
	public function getReactionsByActorForMessages(Participant $participant, array $messageIds): array {
```

**Review by vitormattos — CHANGES_REQUESTED** [review]

File: lib/Controller/ChatController.php (hunk 1 above)
Comment (nickvergessen) [reply lib/Controller/ChatController.php:542]: I dislike this pattern of modifying local content in submethods in 3rd level code.
I'm fine with moving it to a method in the controller, but it's too unique to this place (like the parent loading just before) to put it somewhere else.

Comment (vitormattos) [reply lib/Controller/ChatController.php:542]: Move is good to make more readable the code and understood the main purpose of methods.
I suggested to move to ReactionManager only to maintain reactions code in the same place, but, no problem to maintain in a private method of ChatController

**Review by vitormattos — APPROVED** [review]

**Pull request merged**